### PR TITLE
[VI-252] MAP SecurityToken duration validation

### PR DESCRIPTION
**Pull request by bramleyjl**
## Summary

- Updates MAP STS service to validate the expiration time of the returned token; tokens valid for more than `900` seconds are considered invalid and will result in a `502 bad gateway` error if requested through the MAP services controller.

## Related issue(s)

- https://jira.devops.va.gov/browse/VI-252

## Testing done

- [x] *New code is covered by unit tests*
- Create an STS token for a MAP token request & make the request: `locahost:3000/v0/map_services/chatbot/token` - you should receive a 200 response & a token
- Testing behavior on localhost is controlled through vets-api-mockdata file - `vets-api-mockdata/map/secure_token_service/token.yml`
- Update the mockdata `expires_in` value to greater than 900 & re-request the token - you should receive a 502 Bad Gateway error

## Screenshots
image

## What areas of the site does it impact?
MAP STS requests

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/map/security_token/configuration.rb [hunk 1]
```diff
@@ -66,6 +66,10 @@ def client_assertion_patient_id_type
         'icn'
       end
 
+      def max_token_duration
+        900
```
Comment (rileyanderson) [inline lib/map/security_token/configuration.rb:70]: could we make this `900.seconds`? just so it's clear

File: lib/map/security_token/service.rb [hunk 2]
```diff
@@ -57,11 +57,17 @@ def parse_and_raise_error(e, icn, application)
...
+          raise Errors::InvalidTokenDurationError,
+                "#{config.logging_prefix} token failed, token duration exceeds maximum"
+        end
 
         {
           access_token: response_body['access_token'],
           expiration: Time.zone.now + response_body['expires_in']
         }
+      rescue Errors::InvalidTokenDurationError => e
+        raise e
```
Comment (rileyanderson) [inline lib/map/security_token/service.rb:70]: I dont think you need to rescue and re-raise here. This should get rescued in the `token` method

File: lib/map/security_token/service.rb [hunk 3]
```diff
@@ -57,11 +57,17 @@ def parse_and_raise_error(e, icn, application)
 
       def parse_response(response, application, icn)
         response_body = response.body
+        if response_body['expires_in'].to_i > config.max_token_duration
+          raise Errors::InvalidTokenDurationError,
+                "#{config.logging_prefix} token failed, token duration exceeds maximum"
+        end
```
Comment (rileyanderson) [inline lib/map/security_token/service.rb:63]: should we move this check to a separate method? I think it would make it a bit easier to read

**Review by ericboehs — APPROVED** [review]

**Comment by bramleyjl** [comment]
Scope has changed enough that I'm closing this ticket and opening a different PR here: https://github.com/department-of-veterans-affairs/vets-api/pull/19907

**Pull request closed without merging**